src/components/tiles/table/table-tile.tsx

```diff
@@ -95,11 +94,9 @@ const TableToolComponent: React.FC<ITileProps> = observer(function TableToolComp
   // and then looks it up in the background, adds to cache, and updates state to force a refresh.
   const lookupImage = useCallback((value: string) => {
     if (gImageMap.isImageUrl(value)) {
-      let imagePromise = imagePromises.get(value);
-      if (!imagePromise) {
-        imagePromise = gImageMap.getImage(value);
-        imagePromises.set(value, imagePromise);
-        imagePromise.then((image) => {
+      const imageEntry = gImageMap.getCachedImage(value);
+      if (!imageEntry) {
```

Comment:
If the image had already been cached (e.g. because it's used elsewhere in the document) when we get here, then it will never be loaded. You could add an else clause but then you're potentially back to the infinite loop situation. It seems to me you still need some kind of local state to keep track of which image urls have been considered locally.

Maybe you could simplify it to something like this:
```typescript
        gImageMap.getImage(value).then((image) => {
          if (image?.displayUrl && imageUrls.get(value) !== image?.displayUrl) {
            // This state changes triggers a re-render
            setImageUrls(urls => new Map(urls).set(value, image.displayUrl));
          }
        });